oracle/pkg/updater/updater.go

```diff
@@ -336,19 +336,30 @@ func (u *Updater) handleOpenedCommitment(
 	)
 
 	commitmentTxnHashes := strings.Split(update.TxnHash, ",")
+	u.logger.Debug("commitmentTxnHashes", "commitmentTxnHashes", commitmentTxnHashes)
+	revertableTxns := strings.Split(update.RevertingTxHashes, ",")
+	u.logger.Debug("revertableTxns", "revertableTxns", revertableTxns)
+
+	// Create a map for revertable transactions
+	revertableTxnsMap := make(map[string]bool)
+	for _, txn := range revertableTxns {
+		revertableTxnsMap[txn] = true
+	}
+
 	// Ensure Bundle is atomic and present in the block
 	for i := 0; i < len(commitmentTxnHashes); i++ {
 		txnDetails, found := txns[commitmentTxnHashes[i]]
-		if !found || txnDetails.PosInBlock != (txns[commitmentTxnHashes[0]].PosInBlock)+i || !txnDetails.Succeeded {
+		if !found || txnDetails.PosInBlock != (txns[commitmentTxnHashes[0]].PosInBlock)+i || (!txnDetails.Succeeded && !revertableTxnsMap[commitmentTxnHashes[i]]) {
```

Comment:
This is a lot of logic to pack into a single line, consider using multiple ifs for readability

Reply:
Going to keep this for now, because I had it split into multiple Ifs and other reviewers didn't like it.